app/src/main/java/com/example/weather/screen/home/WeatherFragment.kt

```diff
@@ -78,29 +108,60 @@ class WeatherFragment private constructor() : BaseFragment<FragmentWeatherBindin
     }
 
     override fun onGetCurrentWeatherSuccess(weather: Weather) {
-        mWeatherTemp = weather
+        if (mWeatherTemp == null) {
+            mWeatherTemp = weather
+        }
         bindDataToView(weather)
     }
 
     private fun bindDataToView(weather: Weather) {
+        if (weather.isFavorite == "false") {
+            viewBinding.layoutHeader.locationIcon.visibility = View.VISIBLE
+        } else {
+            viewBinding.layoutHeader.locationIcon.visibility = View.GONE
+        }
+
         weather.weatherCurrent?.let {
             val time = it.dateTime?.unixTimestampToTimeString()?.toInt()
             if (time != null) {
                 getIcon(it.weatherMainCondition!!, time)?.let { image ->
                     viewBinding.icWeather.setImageResource(image)
                 }
             }
-
             viewBinding.layoutWeatherBasic.tvDateTime.text =
                 "Today, " + it.dateTime?.unixTimestampToDateTimeString()
             viewBinding.layoutWeatherBasic.tvTemperature.text = it.temperature?.kelvinToCelsius().toString()
-            viewBinding.layoutWeatherBasic.tvDescription.text = it.weatherDescription
+            viewBinding.layoutWeatherBasic.tvDescription.text = weather.city
             viewBinding.layoutWeatherBasic.layoutBasicDetail.tvWindValue.text =
                 it.windSpeed?.mpsToKmph().toString() + " km/h"
             viewBinding.layoutWeatherBasic.layoutBasicDetail.tvHumidityValue.text = it.humidity.toString() + " %"
         }
     }
 
+    override fun onItemSelected(parent: AdapterView<*>, view: View?, pos: Int, id: Long) {
+        if (mSpinnerCheck++ > 0) { // fix spinner auto click at default position (pos = 0) when it initialization
+            mPosition = pos
+            val itemLatitude = mWeatherList[pos].latitude!!
+            val itemLongitude = mWeatherList[pos].longitude!!
```

Comment:
```suggestion
            val itemLatitude = mWeatherList[pos].latitude!!
            val itemLongitude = mWeatherList[pos].longitude!!
```
Không force unwrap như này crack như chơi, check !null trc khi sử dụng